explainaboard/explainaboard_main.py

```diff
@@ -16,21 +15,30 @@
 from explainaboard.analyzers.draw_hist import draw_bar_chart_from_reports
 from explainaboard.constants import Source
 from explainaboard.info import SysOutputInfo
-from explainaboard.loaders.file_loader import DatalabLoaderOption
+from explainaboard.loaders.file_loader import (
+    DatalabLoaderOption,
+    FileLoaderField,
+    FileLoaderMetadata,
+)
 from explainaboard.metric import metric_name_to_config
 from explainaboard.utils.logging import get_logger
 from explainaboard.utils.tensor_analysis import (
     aggregate_score_tensor,
     filter_score_tensor,
     print_score_tensor,
 )
+from explainaboard.utils.typing_utils import unwrap
 
 
-def get_tasks(task: str, system_outputs: list[str]) -> list[str]:
-    """TODO: TaskType.text_classification is set for temporal use, this needs to be
-    fixed
+def get_tasks(task: TaskType, system_outputs: list[str]) -> list[TaskType]:
+    """
+    Get the task for each system output.
+    :param task: Explicitly specified task. Use if present
+    :param system_outputs: System output files, load from metadata in these files if
+      an explicit task is not set
+    :return: A list of task types for each system
     """
-    real_tasks: list[str]
+    real_tasks: list[TaskType] = []
```

Comment:
We all realize that it's probably a little stupid to get tasks (using a dummy task) in this way, maybe in the future we could think about some elegant way to solve this issue fundamentally.

Reply:
I think that I actually solved it by just reading in the JSON without using a loader?

Reply:
>"Or did you mean you wanted to solve it without reading the JSON twice?"

yes, maybe it's impossible if we don't introduce any more arguments.